### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by abihf**

### CODE REVIEW

File: examples/simple/main.tf [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module "test" {
```
Comment (andysaputra) [inline examples/simple/main.tf]: Please give a more realistic example (like using .tvlk.cloud)

File: variables.tf [hunk 2]
```diff
@@ -0,0 +1,41 @@
+variable "name" {
+  type        = "string"
+  description = "name of the hosted zone"
```
Comment (andysaputra) [inline variables.tf]: Sentence case please

File: variables.tf [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+variable "associated_vpcs" {
+  type        = "list"
+  default     = []
+  description = "list of VPCs that will also be associated with this zone"
+}
+
+variable "force_destroy" {
+  default     = false
+  description = "whether to destroy all records inside if the hosted zone is deleted"
```
Comment (andysaputra) [inline variables.tf]: Sentence case please

File: variables.tf [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+variable "main_vpc" {
+  type        = "string"
+  description = "main VPC ID that will be associated with this hosted zone"
+}
+
+variable "associated_vpcs" {
+  type        = "list"
+  default     = []
+  description = "list of VPCs that will also be associated with this zone"
```
Comment (andysaputra) [inline variables.tf]: Sentence case please

File: variables.tf [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+variable "force_destroy" {
+  default     = false
+  description = "whether to destroy all records inside if the hosted zone is deleted"
+}
+
+variable "tags" {
+  default     = {}
+  type        = "map"
+  description = "tags map for the hoosted zone"
```
Comment (andysaputra) [inline variables.tf]: Sentence case please

File: variables.tf [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+variable "environment" {
+  type        = "string"
+  description = "Environment this Route 53 zone belongs to"
+}
+
+variable "main_vpc" {
+  type        = "string"
+  description = "main VPC ID that will be associated with this hosted zone"
```
Comment (andysaputra) [inline variables.tf]: Sentence case please

File: outputs.tf [hunk 7]
```diff
@@ -0,0 +1,4 @@
+output "zone_id" {
+  value       = "${aws_route53_zone.main.zone_id}"
+  description = "hosted zone id"
```
Comment (andysaputra) [inline outputs.tf]: Sentence case please

File: variables.tf [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+variable "associated_vpcs" {
```
Comment (andysaputra) [inline variables.tf]: Should this be called `secondary_vpcs` instead? Or something which indicate that the user should not include the main_vpc here.

File: examples/simple/main.tf [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+  source = "../.."
+
+  name            = "example.domain.private"
```
Comment (andysaputra) [inline examples/simple/main.tf]: Can we provide a more realistic example?

File: outputs.tf (hunk 7 above)
Comment (andysaputra) [reply outputs.tf]: Change to `The hosted zone id`

File: variables.tf [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+variable "secondary_vpcs" {
+  type        = "list"
+  default     = []
+  description = "List of VPCs that will also be associated with this zone"
+}
+
+variable "force_destroy" {
+  default     = false
```
Comment (andysaputra) [inline variables.tf]: missing type

File: examples/simple/main.tf [hunk 11]
```diff
@@ -0,0 +1,11 @@
+module "test" {
+  source = "../.."
+
+  name           = "fpr.traveloka.com"
```
Comment (salvianreynaldi) [inline examples/simple/main.tf]: Will we use `.traveloka.com` for private hosted zones?

File: variables.tf [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+  type        = "string"
```
Comment (salvianreynaldi) [inline variables.tf]: let's make the order consistent, `type`, `default`, `description`

**Pull request merged**